routers/repo/download.go

```diff
@@ -111,12 +128,22 @@ func ServeBlobOrLFS(ctx *context.Context, blob *git.Blob) error {
 				log.Error("ServeBlobOrLFS: Close: %v", err)
 			}
 		}()
-		return ServeData(ctx, ctx.Repo.TreePath, meta.Size, lfsDataRc)
+		return ServeData(ctx, etag, ctx.Repo.TreePath, meta.Size, lfsDataRc)
 	}
 
 	return ServeBlob(ctx, blob)
 }
 
+// handleETag returns true if the etag matches the request and sends StatusNotModified
+func handleETag(ctx *context.Context, etag string) bool {
+	if ctx.Req.Header.Get("If-None-Match") == etag {
+		ctx.Resp.Header().Set("Etag", etag)
+		ctx.Resp.WriteHeader(http.StatusNotModified)
+		return true
+	}
+	return false
+}
+
```

Comment:
Can we re-use `httpcache.HandleEtagCache` instead of duplicating here?

Reply:
Attachments are immutable...? The `UUID` is unique. I will move the logic into the httpcache file.

Reply:
I meant UUID is a bad ETag because content can change on the storage, better would be file modification time 😉

Reply:
I haven't found a method which updates the attachment content after it was created. So the content is bound to the UUID I guess?

Reply:
If the content could get modified on the external storage, then it is possible too that the modified timestamp does not change.
Anyway, the reason for this PR was to reduce server load. To get `os.FileInfo`-data from git blobs requires more work than relying on what the blob id gives us for free (or the attachments uuid).

Reply:
Yeah okay, current approach is fine with me.